src/routes/users/index.ts

```diff
@@ -243,4 +244,21 @@ router.post<Express.EmptyParams, { token: string }, { role: Roles }, void, Requi
     })
 );
 
+type IntrospectParams = {
+    token: string;
+};
+
+router.get<IntrospectParams, UserInfo, void, void>(
+    '/introspect/:token',
+    makeEndpoint(async (req, res) => {
+        const { token } = req.params;
+        const user = await getUserWithToken(token);
+        // res.status(200).send();
+        res.status(200).send({
+            ...filterSensitiveData(user),
+            settings: user.settings,
+        });
```

Comment:
Add  a cookie here, because all of our other protected endpoints will fail if there's no cookie -- refer to /login endpoint to see how to set a cookie (it's a few lines of code).